### ISSUE TITLE
Loss is inf,when training detect model 

### ISSUE THREAD

**Issue by JasonFinisher123** [ultralytics/ultralytics#401]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

cls_loss is becoming inf ,when training detect model ?

**Comment by JasonFinisher123** [comment]
and it is work ,when i training that dataset by Yolov5. so ,is it something wrong?

**Comment by gloomyfish1998** [comment]
i got the same issue on yolov8 segment training while yolov5 segment training work fine!

**Comment by gloomyfish1998** [comment]
D:\python\my_yolov8_train_demo>yolo task=segment mode=train model=yolov8s-seg.pt epochs=100 batch=1 data=datasets/co2_training/dataset.yaml lr0=0.005 lrf=0.005
Ultralytics YOLOv8.0.8  Python-3.8.5 torch-1.12.1+cu113 CUDA:0 (NVIDIA GeForce RTX 3050 Ti Laptop GPU, 4096MiB)
yolo\engine\trainer: task=segment, mode=train, model=yolov8s-seg.pt, data=datasets/co2_training/dataset.yaml, epochs=100, patience=50, batch=1, imgsz=640, save=True, cache=False, device=, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=False, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, retina_masks=False, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=17, workspace=4, nms=False, lr0=0.005, lrf=0.005, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, hydra={'output_subdir': None, 'run': {'dir': '.'}}, v5loader=False, save_dir=runs\segment\train2
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1       928  ultralytics.nn.modules.Conv                  [3, 32, 3, 2]
  1                  -1  1     18560  ultralytics.nn.modules.Conv                  [32, 64, 3, 2]
  2                  -1  1     29056  ultralytics.nn.modules.C2f                   [64, 64, 1, True]
  3                  -1  1     73984  ultralytics.nn.modules.Conv                  [64, 128, 3, 2]
  4                  -1  2    197632  ultralytics.nn.modules.C2f                   [128, 128, 2, True]
  5                  -1  1    295424  ultralytics.nn.modules.Conv                  [128, 256, 3, 2]
  6                  -1  2    788480  ultralytics.nn.modules.C2f                   [256, 256, 2, True]
  7                  -1  1   1180672  ultralytics.nn.modules.Conv                  [256, 512, 3, 2]
  8                  -1  1   1838080  ultralytics.nn.modules.C2f                   [512, 512, 1, True]
  9                  -1  1    656896  ultralytics.nn.modules.SPPF                  [512, 512, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.Concat                [1]
 12                  -1  1    591360  ultralytics.nn.modules.C2f                   [768, 256, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.Concat                [1]
 15                  -1  1    148224  ultralytics.nn.modules.C2f                   [384, 128, 1]
 16                  -1  1    147712  ultralytics.nn.modules.Conv                  [128, 128, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.Concat                [1]
 18                  -1  1    493056  ultralytics.nn.modules.C2f                   [384, 256, 1]
 19                  -1  1    590336  ultralytics.nn.modules.Conv                  [256, 256, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.Concat                [1]
 21                  -1  1   1969152  ultralytics.nn.modules.C2f                   [768, 512, 1]
 22        [15, 18, 21]  1   2770931  ultralytics.nn.modules.Segment               [1, 32, 128, [128, 256, 512]]
YOLOv8s-seg summary: 261 layers, 11790483 parameters, 11790467 gradients, 42.7 GFLOPs

Transferred 411/417 items from pretrained weights
optimizer: SGD(lr=0.005) with parameter groups 66 weight(decay=0.0), 77 weight(decay=0.0005), 76 bias
train: Scanning D:\python\my_yolov8_train_demo\datasets\co2_training\data\labels\train.cache... 95 images, 0 backgrounds, 0 corrupt: 100%|█
val: Scanning D:\python\my_yolov8_train_demo\datasets\co2_training\data\labels\valid.cache... 6 images, 0 backgrounds, 0 corrupt: 100%|████
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs\segment\train2
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      1/100     0.472G      3.732      9.855        inf      2.154          1        640: 100%|██████████| 95/95 [00:10<00:00,  9.22it/s]
c:\users\administrator\appdata\local\programs\python\python38\lib\site-packages\torch\optim\lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|
                   all          6          6          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      2/100     0.556G      4.041      9.073        inf        2.3          1        640: 100%|██████████| 95/95 [00:08<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|
                   all          6          6          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      3/100     0.558G      3.358      9.139        inf      2.103          1        640:  68%|██████▊   | 65/95 [00:05<00:02, 11.32it/s]

**Comment by pderrenger** [comment]
@gloomyfish1998 it seems like you are experiencing an issue with an infinite `cls_loss` during training of the YOLOv8s-seg model for the segmentation task.

There are a few potential reasons for this issue:
1. **Incorrect Labeling**: Ensure that the labels in your dataset are correctly formatted and match the classes specified in your model configuration file.

2. **Data Imbalance**: Check for class imbalance in your dataset. If one or more classes are underrepresented, it can lead to convergence issues.

3. **Learning Rate**: Try adjusting the learning rate (`lr0` and `lrf`) to see if it has an impact on the convergence behavior.

4. **Model Architecture**: Review the architecture of YOLOv8s-seg and consider if it's suitable for your specific segmentation task. You may want to experiment with different model architectures or hyperparameters.

I recommend double-checking your dataset, ensuring label correctness, addressing any data imbalances, and experimenting with different learning rates or model architectures to troubleshoot this issue.

If you're still encountering the problem, feel free to share more details about your dataset and training setup so that I can assist you further!

**Comment by npn-zakipoint** [comment]
@pderrenger , I ended up with inf in the box_loss. Currently I am training the YOLO v11 medium model on PubMed dataset. 

How it is possible to get the inf as loss if there are imbalance data & model architecture. I am confused here. Could you please add more intuitive explanation to support that statement?

I have the following box_loss during the training:


```
Epoch 1: 0.39842  
Epoch 2: inf  
Epoch 3: 0.39733  
Epoch 4: 0.39683  
Epoch 5: inf  
Epoch 6: inf  
Epoch 7: 0.39642  
Epoch 8: inf  
Epoch 9: 0.39593  
Epoch 10: 0.39514 
```

Thank you in advance.

**Comment by Y-T-G** [comment]
@npn-zakipoint What was the training command you used?

**Comment by krisssnv** [comment]
I have a similar issue with yolo v11. I checked several times the data and it seems there are no label mismatches, yet whatever hyperparameters I use I always get a box loss = inf. For example these are the values from my last train attempt:
epoch 1: 0.389, 
epoch 2: 0.3417, 
epoch 3: 0.3237, 
epoch 4: 0.312, 
epoch 5: inf;
Could you please provide more details as to how and why this could be happening? Thank you ! 

PS has anyone managed to resolve it?

**Comment by Y-T-G** [comment]
@krisssnv If you're setting `optimizer`, then remove it.
